src/interp/interp-wasi.cc

```diff
@@ -753,7 +756,7 @@ Result WasiRunStart(const Instance::Ptr& instance,
   }
 
   // Unregister memory
-  wasiInstances.erase(instance.get());
+  UnregisterWasiInstance(instance);
```

Comment:
This needs to be renamed I think.

Reply:
Should this line be removed now that the registration and unregistration now all seems to be taken care in wasm-interp.cc?

Reply:
Nope, You might use it elsewhere (in spec-test for example). I did not want to monolith `wasm-interp.cc`.